bitwarden: 2022.8.1 -> 2022.10.0

kilianar opened the pull request:
###### Description of changes
- https://github.com/bitwarden/clients/releases/tag/desktop-v2022.9.0
- https://github.com/bitwarden/clients/releases/tag/desktop-v2022.9.1
- https://github.com/bitwarden/clients/releases/tag/desktop-v2022.10.0

Install desktopItem with `copyDesktopItems`.

###### Things done
- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

kilianar commented:
Result of `nixpkgs-review pr 190951` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>bitwarden</li>
  </ul>
</details>

ajs124 commented:
This leads to the `.desktop` file containing `Exec=/opt/Bitwarden/bitwarden %U`, which is probably wrong

kilianar committed:
bitwarden: 2022.8.1 -> 2022.10.0

- https://github.com/bitwarden/clients/releases/tag/desktop-v2022.9.0
- https://github.com/bitwarden/clients/releases/tag/desktop-v2022.9.1
- https://github.com/bitwarden/clients/releases/tag/desktop-v2022.10.0

kilianar commented:
You're right. It didn't copy the desktop item from `desktopItems` but used the file already installed in `$out/share/applications/`. I reverted the `dektopItem` related changed and updated to [version 2022.10.0](https://github.com/bitwarden/clients/releases/tag/desktop-v2022.10.0).
